In [2]:
import numpy as np
import pandas as pd
import re

import matplotlib.pyplot as plt
import seaborn as sns

# **Acceso a Internet**

A continuación, cargamos, exploramos y analizamos los datasets proporcionados por [ENACOM](https://datosabiertos.enacom.gob.ar/dashboards/20000/acceso-a-internet/) y guardadas en la carpeta [Original datasets] correspondientes a la información del mercado, oferta, demanda y cobertura del servicio de internet (Primer Trimestre 2014 - Tercer Trimestre 2022)

## Penetración de Internet fijo (accesos por cada 100 hogares) **`penetracion_hogar_prov`**

Número de accesos al servicio de Internet fijo por cada 100 hogares por provincia

In [17]:
penetracion_hogar_prov = pd.read_csv('../Telecommunications data analysis/Original datasets/Internet_Penetracion.csv')
penetracion_hogar_prov

Año  Trimestre            Provincia Accesos por cada 100 hogares
0    2022          4         Buenos Aires                        78,94
1    2022          4      Capital Federal                       122,73
2    2022          4            Catamarca                        65,77
3    2022          4                Chaco                        43,62
4    2022          4               Chubut                        85,57
..    ...        ...                  ...                          ...
859  2014          1           Santa Cruz                        28,94
860  2014          1             Santa Fe                        47,85
861  2014          1  Santiago Del Estero                        16,31
862  2014          1     Tierra Del Fuego                        63,97
863  2014          1              Tucumán                        33,42

[864 rows x 4 columns]

Mostramos la información del dataframe

In [18]:
penetracion_hogar_prov.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 864 entries, 0 to 863
Data columns (total 4 columns):
 #   Column                        Non-Null Count  Dtype 
---  ------                        --------------  ----- 
 0   Año                           864 non-null    int64 
 1   Trimestre                     864 non-null    int64 
 2   Provincia                     864 non-null    object
 3   Accesos por cada 100 hogares  864 non-null    object
dtypes: int64(2), object(2)
memory usage: 27.1+ KB


La columna **Accesos por cada 100 hogares** es de tipo cadena cuando debe ser de tipo numérico

In [19]:
# Reemplazamos las comas por puntos para que el método de conversión no encuentre errores
penetracion_hogar_prov['Accesos por cada 100 hogares'] = penetracion_hogar_prov['Accesos por cada 100 hogares'].str.replace(',', '.')
# Convertimos la columna a numérico
penetracion_hogar_prov['Accesos por cada 100 hogares'] = pd.to_numeric(penetracion_hogar_prov['Accesos por cada 100 hogares'])
# Mostramos la información actualizada del dataframe
penetracion_hogar_prov.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 864 entries, 0 to 863
Data columns (total 4 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Año                           864 non-null    int64  
 1   Trimestre                     864 non-null    int64  
 2   Provincia                     864 non-null    object 
 3   Accesos por cada 100 hogares  864 non-null    float64
dtypes: float64(1), int64(2), object(1)
memory usage: 27.1+ KB


Verificamos las provincias

In [21]:
penetracion_hogar_prov['Provincia'].unique()

array(['Buenos Aires', 'Capital Federal', 'Catamarca', 'Chaco', 'Chubut',
       'Córdoba', 'Corrientes', 'Entre Ríos', 'Formosa', 'Jujuy',
       'La Pampa', 'La Rioja', 'Mendoza', 'Misiones', 'Neuquén',
       'Río Negro', 'Salta', 'San Juan', 'San Luis', 'Santa Cruz',
       'Santa Fe', 'Santiago Del Estero', 'Tierra Del Fuego', 'Tucumán'],
      dtype=object)

Resumen estadístico

In [22]:
penetracion_hogar_prov.describe()

Año   Trimestre  Accesos por cada 100 hogares
count   864.000000  864.000000                    864.000000
mean   2018.000000    2.500000                     49.620301
std       2.583484    1.118682                     23.319150
min    2014.000000    1.000000                      9.350000
25%    2016.000000    1.750000                     32.220000
50%    2018.000000    2.500000                     46.390000
75%    2020.000000    3.250000                     62.442500
max    2022.000000    4.000000                    124.060000

**Observaciones**

- La media de accesos por cada 100 hogares es de 49.62, lo que indica que hay menos de 50 accesos a internet (en promedio) en Argentina.
- El valor mínimo de Accesos es 9.35 lo que indica lo que indica que existen provincias con una penetración de internet muy baja. Al mismo tiempo, observamos que el valor máximo es de 124.06, lo que indica que hay algunas provincias con muy alta penetración de internet por hogares.

Análisis Univariable

## **`internet_por_tech_loc`**
Número de accesos al servicio de Internet fijo por tecnología en cada localidad declarada

In [3]:
internet_por_tech_loc = pd.read_csv('../Telecommunications data analysis/Original datasets/Accesos_a_Internet_fijo_por_tecnologia_y_localidad.csv')
internet_por_tech_loc

Provincia      Partido                           Localidad  \
0     BUENOS AIRES   25 de Mayo                          25 de Mayo   
1     BUENOS AIRES   25 de Mayo                           Del Valle   
2     BUENOS AIRES   25 de Mayo                   Gobernador Ugarte   
3     BUENOS AIRES   25 de Mayo              Norberto de la Riestra   
4     BUENOS AIRES   25 de Mayo                    Lucas Monteverde   
...            ...          ...                                 ...   
3132       TUCUMAN   Tafí Viejo  Villa Mariano Moreno - El Colmenar   
3133       TUCUMAN      Trancas                San Pedro de Colalao   
3134       TUCUMAN      Trancas                   Villa  de Trancas   
3135       TUCUMAN  Yerba Buena                       Villa Carmela   
3136       TUCUMAN  Yerba Buena            Yerba Buena - Marcos Paz   

     Link Indec ADSL CABLEMODEM DIAL UP FIBRA OPTICA OTROS SATELITAL WIMAX  \
0       6854100  874       4,54     - 0            3   - 0       852   - 0   
1       6854020  181        - 0     - 0          - 0    10         1   - 0   
2       6854040  - 0        - 0     - 0          - 0   - 0       - 0   - 0   
3       6854060  - 0        782     - 0            6   167       - 0   - 0   
4       6854050  - 0        - 0     - 0          - 0   - 0       - 0   - 0   
...         ...  ...        ...     ...          ...   ...       ...   ...   
3132   90105100  - 0         31     - 0         3,73   - 0       - 0   - 0   
3133   90112020  - 0        - 0     - 0          - 0   - 0       - 0   - 0   
3134   90112030  126        - 0     - 0          - 0    33       - 0   - 0   
3135   90119020  1,5        - 0     - 0          - 0     1       - 0   - 0   
3136   90119030  129        209     - 0         15,4    33       - 0   - 0   

     WIRELESS Total general  Unnamed: 13  
0         706          6,98          NaN  
1         - 0           192          NaN  
2         181           181          NaN  
3         327          1,28          NaN  
4           6             6          NaN  
...       ...           ...          ...  
3132      - 0          3,76          NaN  
3133      313           313          NaN  
3134      101           260          NaN  
3135       97           1,6          NaN  
3136      - 0         15,78          NaN  

[3137 rows x 14 columns]

Eliminamos la última columna

In [4]:
internet_por_tech_loc = internet_por_tech_loc.iloc[:, 0:13]
internet_por_tech_loc.head()

Provincia     Partido               Localidad Link Indec ADSL  \
0  BUENOS AIRES  25 de Mayo              25 de Mayo    6854100  874   
1  BUENOS AIRES  25 de Mayo               Del Valle    6854020  181   
2  BUENOS AIRES  25 de Mayo       Gobernador Ugarte    6854040  - 0   
3  BUENOS AIRES  25 de Mayo  Norberto de la Riestra    6854060  - 0   
4  BUENOS AIRES  25 de Mayo        Lucas Monteverde    6854050  - 0   

  CABLEMODEM DIAL UP FIBRA OPTICA OTROS SATELITAL WIMAX WIRELESS Total general  
0       4,54     - 0            3   - 0       852   - 0      706          6,98  
1        - 0     - 0          - 0    10         1   - 0      - 0           192  
2        - 0     - 0          - 0   - 0       - 0   - 0      181           181  
3        782     - 0            6   167       - 0   - 0      327          1,28  
4        - 0     - 0          - 0   - 0       - 0   - 0        6             6

Reemplazamos valores

In [5]:
# Para las columnas a partir de la quinta, eliminamos los guiones, y las comas lo reemplazamos por puntos

for columna in internet_por_tech_loc.columns[4:]:
    internet_por_tech_loc[columna] = internet_por_tech_loc[columna].str.replace('- ', '')
    internet_por_tech_loc[columna] = internet_por_tech_loc[columna].str.replace(',', '.')

internet_por_tech_loc.head()

Provincia     Partido               Localidad Link Indec ADSL  \
0  BUENOS AIRES  25 de Mayo              25 de Mayo    6854100  874   
1  BUENOS AIRES  25 de Mayo               Del Valle    6854020  181   
2  BUENOS AIRES  25 de Mayo       Gobernador Ugarte    6854040    0   
3  BUENOS AIRES  25 de Mayo  Norberto de la Riestra    6854060    0   
4  BUENOS AIRES  25 de Mayo        Lucas Monteverde    6854050    0   

  CABLEMODEM DIAL UP FIBRA OPTICA OTROS SATELITAL WIMAX WIRELESS Total general  
0       4.54       0            3     0       852     0      706          6.98  
1          0       0            0    10         1     0        0           192  
2          0       0            0     0         0     0      181           181  
3        782       0            6   167         0     0      327          1.28  
4          0       0            0     0         0     0        6             6

Cambiamos el tipo de dato

In [6]:
for columna in internet_por_tech_loc.columns[4:]:
    internet_por_tech_loc[columna] =pd.to_numeric(internet_por_tech_loc[columna], errors='coerce').fillna(0)

internet_por_tech_loc.head()

Provincia     Partido               Localidad Link Indec   ADSL  \
0  BUENOS AIRES  25 de Mayo              25 de Mayo    6854100  874.0   
1  BUENOS AIRES  25 de Mayo               Del Valle    6854020  181.0   
2  BUENOS AIRES  25 de Mayo       Gobernador Ugarte    6854040    0.0   
3  BUENOS AIRES  25 de Mayo  Norberto de la Riestra    6854060    0.0   
4  BUENOS AIRES  25 de Mayo        Lucas Monteverde    6854050    0.0   

   CABLEMODEM  DIAL UP  FIBRA OPTICA  OTROS  SATELITAL  WIMAX  WIRELESS  \
0        4.54      0.0           3.0    0.0        852      0     706.0   
1        0.00      0.0           0.0   10.0          1      0       0.0   
2        0.00      0.0           0.0    0.0          0      0     181.0   
3      782.00      0.0           6.0  167.0          0      0     327.0   
4        0.00      0.0           0.0    0.0          0      0       6.0   

   Total general  
0           6.98  
1         192.00  
2         181.00  
3           1.28  
4           6.00

In [7]:
internet_por_tech_loc.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3137 entries, 0 to 3136
Data columns (total 13 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Provincia      3137 non-null   object 
 1   Partido        3137 non-null   object 
 2   Localidad      3137 non-null   object 
 3   Link Indec     3137 non-null   object 
 4   ADSL           3137 non-null   float64
 5   CABLEMODEM     3137 non-null   float64
 6   DIAL UP        3137 non-null   float64
 7   FIBRA OPTICA   3137 non-null   float64
 8   OTROS          3137 non-null   float64
 9   SATELITAL      3137 non-null   int64  
 10  WIMAX          3137 non-null   int64  
 11  WIRELESS       3137 non-null   float64
 12  Total general  3137 non-null   float64
dtypes: float64(7), int64(2), object(4)
memory usage: 318.7+ KB


Verificamos la columna **Provincia**

In [8]:
internet_por_tech_loc['Provincia'].unique()

array(['BUENOS AIRES', 'CABA', 'CATAMARCA', 'CHACO', 'CHUBUT', 'CORDOBA',
       'CORRIENTES', 'ENTRE RIOS', 'FORMOSA', 'JUJUY', 'LA PAMPA',
       'LA RIOJA', 'MENDOZA', 'MISIONES', 'NEUQUEN', 'RIO NEGRO', 'SALTA',
       'SAN JUAN', 'SAN LUIS', 'SANTA CRUZ', 'SANTA FE',
       'SANTIAGO DEL ESTERO', 'TIERRA DEL FUEGO', 'TUCUMAN'], dtype=object)

## **`internet_por_vel_prov`**
Número de accesos al servicio de Internet fijo por velocidad de bajada en cada provincia

In [9]:
internet_por_vel_prov = pd.read_csv('../Telecommunications data analysis/Original datasets/Accesos_a_Internet_fijo_por_velocidad_bajada_y_provincia.csv')
internet_por_vel_prov

Año  Trimestre            Provincia    Otros 0,256 Mbps 0,375 Mbps  \
0    2022.0        4.0         Buenos Aires   97.634         18        - 0   
1    2022.0        4.0      Capital Federal  106.772         31        - 0   
2    2022.0        4.0            Catamarca    2.280        - 0        - 0   
3    2022.0        4.0                Chaco    3.909        - 0        225   
4    2022.0        4.0               Chubut   20.026          8        - 0   
..      ...        ...                  ...      ...        ...        ...   
500  2017.0        4.0             Santa Fe   14.119        - 0        - 0   
501  2017.0        4.0  Santiago Del Estero      - 0        - 0        - 0   
502  2017.0        4.0     Tierra Del Fuego      - 0        - 0        - 0   
503  2017.0        4.0              Tucumán      - 0        - 0        - 0   
504     NaN        NaN                  NaN      NaN        NaN        NaN   

    0,625 Mbps  0,5 Mbps 0,512 Mbps 0,75 Mbps  ... 63 Mbps 64 Mbps 65 Mbps  \
0          - 0    28.724        192     4.611  ...     - 0     - 0      14   
1          - 0   451.000         34     1.307  ...     - 0     - 0     - 0   
2          - 0    34.000         37       - 0  ...     - 0     - 0     - 0   
3          111   234.000          2       - 0  ...     - 0     - 0     - 0   
4          - 0    69.000         32       836  ...     - 0     - 0     - 0   
..         ...       ...        ...       ...  ...     ...     ...     ...   
500        - 0     3.109        - 0     7.637  ...     - 0     - 0     - 0   
501        - 0    41.000        - 0       909  ...     - 0     - 0     - 0   
502        - 0     6.000        - 0       282  ...     - 0     - 0     - 0   
503        - 0   105.000        - 0         1  ...     - 0     - 0     - 0   
504        NaN       NaN        NaN       NaN  ...     NaN     NaN     NaN   

    66 Mbps 67 Mbps 68 Mbps 70 Mbps 71 Mbps  75 Mbps 77 Mbps  
0       - 0     - 0     - 0   5.655     - 0   90.796     - 0  
1         1     - 0     - 0      15     - 0   28.406     - 0  
2       - 0     - 0     - 0     - 0     - 0      - 0     - 0  
3       - 0     - 0     - 0     - 0     - 0      - 0     - 0  
4       - 0     - 0     - 0     - 0     - 0      - 0     - 0  
..      ...     ...     ...     ...     ...      ...     ...  
500     - 0     - 0     - 0     - 0     - 0      - 0     - 0  
501     - 0     - 0     - 0     - 0     - 0      - 0     - 0  
502     - 0     - 0     - 0     - 0     - 0      - 0     - 0  
503     - 0     - 0     - 0     - 0     - 0      - 0     - 0  
504     NaN     NaN     NaN     NaN     NaN      NaN     NaN  

[505 rows x 97 columns]

Eliminamos valores nulos

In [10]:
internet_por_vel_prov.dropna(inplace=True)

In [11]:
# Se observa la información de cada columna
internet_por_vel_prov.info()

<class 'pandas.core.frame.DataFrame'>
Index: 504 entries, 0 to 503
Data columns (total 97 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Año         504 non-null    float64
 1   Trimestre   504 non-null    float64
 2   Provincia   504 non-null    object 
 3   Otros       504 non-null    object 
 4   0,256 Mbps  504 non-null    object 
 5   0,375 Mbps  504 non-null    object 
 6   0,625 Mbps  504 non-null    object 
 7   0,5 Mbps    504 non-null    float64
 8   0,512 Mbps  504 non-null    object 
 9   0,75 Mbps   504 non-null    object 
 10  1 Mbps      504 non-null    object 
 11  1,2 Mbps    504 non-null    object 
 12  1,25 Mbps   504 non-null    object 
 13  1,5 Mbps    504 non-null    object 
 14  2 Mbps      504 non-null    object 
 15  2,2 Mbps    504 non-null    object 
 16  2,5 Mbps    504 non-null    object 
 17  6,7 Mbps    504 non-null    object 
 18  3 Mbps      504 non-null    float64
 19  3,2 Mbps    504 non-null    object

Según la información de la tabla, las columnas **Año** y **Trimestre** son de tipo flotante cuando deben ser de tipo entero. Las columnas de tipos de servicios son de tipo str con tipos flotantes entre ellos. Así que transformamos todos estos tipos de datos mencionados.

In [12]:
# transformamos las columnas Año y Trimestre a entero
internet_por_vel_prov['Año'] = internet_por_vel_prov['Año'].astype(int)
internet_por_vel_prov['Trimestre'] = internet_por_vel_prov['Trimestre'].astype(int)

In [13]:
# Para las columnas; a partir de la cuarta, eliminamos los guiones, y las comas lo reemplazamos por puntos; en aquellas columnas de tipo str
for columna in internet_por_vel_prov.columns[3:]:
    if internet_por_vel_prov[columna].dtype == object:
        internet_por_vel_prov[columna] = internet_por_vel_prov[columna].str.replace('- ', '')
        internet_por_vel_prov[columna] = internet_por_vel_prov[columna].str.replace(',', '.')

In [14]:
# Transformamos las columnas anteriores a tipo numerico
for columna in internet_por_vel_prov.columns[3:]:
    internet_por_vel_prov[columna] =pd.to_numeric(internet_por_vel_prov[columna], errors='coerce').fillna(0)

internet_por_vel_prov.head()

Año  Trimestre        Provincia    Otros  0,256 Mbps  0,375 Mbps  \
0  2022          4     Buenos Aires   97.634          18           0   
1  2022          4  Capital Federal  106.772          31           0   
2  2022          4        Catamarca    2.280           0           0   
3  2022          4            Chaco    3.909           0         225   
4  2022          4           Chubut   20.026           8           0   

   0,625 Mbps  0,5 Mbps  0,512 Mbps  0,75 Mbps  ...  63 Mbps  64 Mbps  \
0           0    28.724         192      4.611  ...        0        0   
1           0   451.000          34      1.307  ...        0        0   
2           0    34.000          37      0.000  ...        0        0   
3         111   234.000           2      0.000  ...        0        0   
4           0    69.000          32    836.000  ...        0        0   

   65 Mbps  66 Mbps  67 Mbps  68 Mbps  70 Mbps  71 Mbps  75 Mbps  77 Mbps  
0       14        0        0        0    5.655        0   90.796        0  
1        0        1        0        0   15.000        0   28.406        0  
2        0        0        0        0    0.000        0    0.000        0  
3        0        0        0        0    0.000        0    0.000        0  
4        0        0        0        0    0.000        0    0.000        0  

[5 rows x 97 columns]

In [15]:
internet_por_vel_prov.info()

<class 'pandas.core.frame.DataFrame'>
Index: 504 entries, 0 to 503
Data columns (total 97 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Año         504 non-null    int32  
 1   Trimestre   504 non-null    int32  
 2   Provincia   504 non-null    object 
 3   Otros       504 non-null    float64
 4   0,256 Mbps  504 non-null    int64  
 5   0,375 Mbps  504 non-null    int64  
 6   0,625 Mbps  504 non-null    int64  
 7   0,5 Mbps    504 non-null    float64
 8   0,512 Mbps  504 non-null    int64  
 9   0,75 Mbps   504 non-null    float64
 10  1 Mbps      504 non-null    float64
 11  1,2 Mbps    504 non-null    int64  
 12  1,25 Mbps   504 non-null    int64  
 13  1,5 Mbps    504 non-null    float64
 14  2 Mbps      504 non-null    float64
 15  2,2 Mbps    504 non-null    int64  
 16  2,5 Mbps    504 non-null    float64
 17  6,7 Mbps    504 non-null    int64  
 18  3 Mbps      504 non-null    float64
 19  3,2 Mbps    504 non-null    int64 

Verificamos la columna **Provincia**

In [16]:
internet_por_vel_prov['Provincia'].unique()

array(['Buenos Aires', 'Capital Federal', 'Catamarca', 'Chaco', 'Chubut',
       'Córdoba', 'Corrientes', 'Entre Ríos', 'Formosa', 'Jujuy',
       'La Pampa', 'La Rioja', 'Mendoza', 'Misiones', 'Neuquén',
       'Río Negro', 'Salta', 'San Juan', 'San Luis', 'Santa Cruz',
       'Santa Fe', 'Santiago Del Estero', 'Tierra Del Fuego', 'Tucumán'],
      dtype=object)

## **`internet_por_vel_loc`**
Número de accesos al servicio de Internet fijo por velocidad de bajada en cada localidad declarada

In [17]:
internet_por_vel_loc = pd.read_csv('../Telecommunications data analysis/Original datasets/Accesos_a_Internet_fijo_por_velocidad_de_bajada_y_localidad.csv')
internet_por_vel_loc

Provincia      Partido                           Localidad  \
0     BUENOS AIRES   25 de Mayo                          25 de Mayo   
1     BUENOS AIRES   25 de Mayo                     Agustín Mosconi   
2     BUENOS AIRES   25 de Mayo                           Del Valle   
3     BUENOS AIRES   25 de Mayo                           Ernestina   
4     BUENOS AIRES   25 de Mayo                   Gobernador Ugarte   
...            ...          ...                                 ...   
3090       TUCUMAN   Tafí Viejo  Villa Mariano Moreno - El Colmenar   
3091       TUCUMAN      Trancas                San Pedro de Colalao   
3092       TUCUMAN      Trancas                   Villa  de Trancas   
3093       TUCUMAN  Yerba Buena                       Villa Carmela   
3094       TUCUMAN  Yerba Buena            Yerba Buena - Marcos Paz   

     Link Indec  Otros  0,256 Mbps  0,375 Mbps  0,5 Mbps  0,512 Mbps  \
0       6854100    NaN         NaN         NaN       2.0         NaN   
1       6854010    NaN         NaN         NaN       NaN         NaN   
2       6854020    NaN         NaN         NaN       1.0         NaN   
3       6854030    NaN         NaN         NaN       NaN         NaN   
4       6854040    NaN         NaN         NaN       NaN         NaN   
...         ...    ...         ...         ...       ...         ...   
3090   90105100    NaN         NaN         NaN       NaN         NaN   
3091   90112020    NaN         NaN         NaN       NaN         NaN   
3092   90112030    4.0         NaN         NaN       NaN         NaN   
3093   90119020    NaN         NaN         NaN       NaN         NaN   
3094   90119030  176.0         NaN         NaN       NaN         NaN   

      0,625 Mbps  ...  61 Mbps  62 Mbps  64 Mbps  65 Mbps  66 Mbps  70 Mbps  \
0            NaN  ...      NaN      NaN      NaN      NaN      NaN      NaN   
1            NaN  ...      NaN      NaN      NaN      NaN      NaN      NaN   
2            NaN  ...      NaN      NaN      NaN      NaN      NaN      NaN   
3            NaN  ...      NaN      NaN      NaN      NaN      NaN      NaN   
4            NaN  ...      NaN      NaN      NaN      1.0      NaN      NaN   
...          ...  ...      ...      ...      ...      ...      ...      ...   
3090         NaN  ...      NaN      NaN      NaN      NaN      NaN      NaN   
3091         NaN  ...      NaN      NaN      NaN      NaN      NaN      NaN   
3092         NaN  ...      NaN      NaN      NaN      NaN      NaN      NaN   
3093         NaN  ...      NaN      NaN      NaN      NaN      NaN      NaN   
3094         NaN  ...      NaN      NaN      NaN      NaN      NaN      NaN   

      75 Mbps  78 Mbps  80 Mbps  83 Mbps  
0         NaN      NaN     12.0      NaN  
1         NaN      NaN      NaN      NaN  
2         NaN      NaN      NaN      NaN  
3         NaN      NaN      NaN      NaN  
4         NaN      NaN      NaN      NaN  
...       ...      ...      ...      ...  
3090      NaN      NaN      NaN      NaN  
3091      NaN      NaN      NaN      NaN  
3092      NaN      NaN      NaN      NaN  
3093      NaN      NaN      NaN      NaN  
3094      NaN      NaN      NaN      NaN  

[3095 rows x 83 columns]

In [18]:
internet_por_vel_loc.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3095 entries, 0 to 3094
Data columns (total 83 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Provincia   3095 non-null   object 
 1   Partido     3095 non-null   object 
 2   Localidad   3095 non-null   object 
 3   Link Indec  3095 non-null   object 
 4   Otros       962 non-null    float64
 5   0,256 Mbps  28 non-null     float64
 6   0,375 Mbps  1 non-null      float64
 7   0,5 Mbps    643 non-null    float64
 8   0,512 Mbps  523 non-null    float64
 9   0,625 Mbps  1 non-null      float64
 10  0,75 Mbps   284 non-null    float64
 11  1 Mbps      891 non-null    float64
 12  1,25 Mbps   1 non-null      float64
 13  1,5 Mbps    15 non-null     float64
 14  2 Mbps      722 non-null    float64
 15  2,2 Mbps    1 non-null      float64
 16  2,5 Mbps    4 non-null      float64
 17  3 Mbps      1351 non-null   float64
 18  3,3 Mbps    1 non-null      float64
 19  3,5 Mbps    373 non-null   

Verificamos la columna **Provincia**

In [19]:
internet_por_vel_loc['Provincia'].unique()

array(['BUENOS AIRES', 'CABA', 'CATAMARCA', 'CHACO', 'CHUBUT', 'CORDOBA',
       'CORRIENTES', 'ENTRE RIOS', 'FORMOSA', 'JUJUY', 'LA PAMPA',
       'LA RIOJA', 'MENDOZA', 'MISIONES', 'NEUQUEN', 'RIO NEGRO', 'SALTA',
       'SAN JUAN', 'SAN LUIS', 'SANTA CRUZ', 'SANTA FE',
       'SANTIAGO DEL ESTERO', 'TIERRA DEL FUEGO', 'TUCUMAN'], dtype=object)

## **`velocidad_internet`**

Serie histórica de la velocidad media de descarga de Internet por provincia (trimestral)

In [20]:
velocidad_internet = pd.read_csv('../Telecommunications data analysis/Original datasets/historico_velocidad_internet.csv')
velocidad_internet

Año  Trimestre            Provincia  Mbps (Media de bajada)
0    2022          4         Buenos Aires                 111.000
1    2022          4      Capital Federal                  18.286
2    2022          4            Catamarca                   8.333
3    2022          4                Chaco                  10.402
4    2022          4               Chubut                   1.592
..    ...        ...                  ...                     ...
859  2014          1           Santa Cruz                 332.000
860  2014          1             Santa Fe                 315.000
861  2014          1  Santiago Del Estero                 295.000
862  2014          1     Tierra Del Fuego                 345.000
863  2014          1              Tucumán                 321.000

[864 rows x 4 columns]

In [21]:
velocidad_internet.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 864 entries, 0 to 863
Data columns (total 4 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Año                     864 non-null    int64  
 1   Trimestre               864 non-null    int64  
 2   Provincia               864 non-null    object 
 3   Mbps (Media de bajada)  864 non-null    float64
dtypes: float64(1), int64(2), object(1)
memory usage: 27.1+ KB


Verificamos la columna **Provincia**

In [22]:
velocidad_internet['Provincia'].unique()

array(['Buenos Aires', 'Capital Federal', 'Catamarca', 'Chaco', 'Chubut',
       'Córdoba', 'Corrientes', 'Entre Ríos', 'Formosa', 'Jujuy',
       'La Pampa', 'La Rioja', 'Mendoza', 'Misiones', 'Neuquén',
       'Río Negro', 'Salta', 'San Juan', 'San Luis', 'Santa Cruz',
       'Santa Fe', 'Santiago Del Estero', 'Tierra Del Fuego', 'Tucumán'],
      dtype=object)

## **`acceso_internet_tec`**

Número de accesos al servicio de Internet fijo por tipo de tecnología en cada provincia (trimestral)

In [23]:
acceso_internet_tec = pd.read_csv('../Telecommunications data analysis/Original datasets/Internet_Accesos_por_tecnologia.csv')
acceso_internet_tec

Año                                          Trimestre  \
0    2022                                                  4   
1    2022                                                  4   
2    2022                                                  4   
3    2022                                                  4   
4    2022                                                  4   
..    ...                                                ...   
860  2014                                                  1   
861  2014                                                  1   
862  2014                                                  1   
863  2014                                                  1   
864     *  Los datos provinciales no coinciden a nivel na...   

               Provincia   ADSL  Cablemodem  Fibra óptica  Wireless  Otros  \
0           Buenos Aires  326.0         3.0           1.0     130.0   72.0   
1        Capital Federal  139.0         1.0         139.0       5.0   32.0   
2              Catamarca    9.0        11.0          47.0       1.0    2.0   
3                  Chaco   25.0        61.0          47.0       8.0    3.0   
4                 Chubut   45.0        74.0          10.0      31.0    9.0   
..                   ...    ...         ...           ...       ...    ...   
860             Santa Fe  323.0       174.0           3.0       6.0  644.0   
861  Santiago Del Estero   33.0         4.0          19.0     915.0   34.0   
862     Tierra Del Fuego   22.0         3.0         648.0       1.0    3.0   
863              Tucumán  130.0        83.0         121.0      13.0   98.0   
864                  NaN    NaN         NaN           NaN       NaN    NaN   

     Total  
0      5.0  
1      2.0  
2     71.0  
3    144.0  
4    169.0  
..     ...  
860  507.0  
861   37.0  
862   28.0  
863  130.0  
864    NaN  

[865 rows x 9 columns]

Eliminamos valores nulos

In [24]:
acceso_internet_tec.dropna(inplace=True)

In [25]:
acceso_internet_tec.info()

<class 'pandas.core.frame.DataFrame'>
Index: 864 entries, 0 to 863
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Año           864 non-null    object 
 1   Trimestre     864 non-null    object 
 2   Provincia     864 non-null    object 
 3   ADSL          864 non-null    float64
 4   Cablemodem    864 non-null    float64
 5   Fibra óptica  864 non-null    float64
 6   Wireless      864 non-null    float64
 7   Otros         864 non-null    float64
 8   Total         864 non-null    float64
dtypes: float64(6), object(3)
memory usage: 67.5+ KB


Verificamos la columna **Provincia**

In [26]:
acceso_internet_tec['Provincia'].unique()

array(['Buenos Aires', 'Capital Federal', 'Catamarca', 'Chaco', 'Chubut',
       'Córdoba', 'Corrientes', 'Entre Ríos', 'Formosa', 'Jujuy',
       'La Pampa', 'La Rioja', 'Mendoza', 'Misiones', 'Neuquén',
       'Río Negro', 'Salta', 'San Juan', 'San Luis', 'Santa Cruz',
       'Santa Fe', 'Santiago Del Estero', 'Tierra Del Fuego', 'Tucumán'],
      dtype=object)

## **`acceso_internet_vel`**

Número de accesos al servicio de Internet fijo por velocidad de bajada en cada provincia (trimestral)

In [27]:
acceso_internet_vel = pd.read_csv('../Telecommunications data analysis/Original datasets/Internet_Accesos-por-velocidad.csv')
acceso_internet_vel

Año  Trimestre            Provincia  HASTA 512 kbps  \
0    2022          4         Buenos Aires          28.934   
1    2022          4      Capital Federal         516.000   
2    2022          4            Catamarca          71.000   
3    2022          4                Chaco         461.000   
4    2022          4               Chubut         109.000   
..    ...        ...                  ...             ...   
859  2014          1           Santa Cruz         161.000   
860  2014          1             Santa Fe           8.456   
861  2014          1  Santiago Del Estero           1.234   
862  2014          1     Tierra Del Fuego          12.000   
863  2014          1              Tucumán           6.000   

     + 512 Kbps - 1 Mbps + 1 Mbps - 6 Mbps  + 6 Mbps - 10 Mbps  \
0                 23.347           259.866             281.550   
1                  5.096            31.320              64.653   
2                315.000             2.716               5.028   
3                771.000            16.067              17.846   
4                  1.614            46.457              31.007   
..                   ...               ...                 ...   
859                1.625            24.972               1.000   
860              124.468           345.225              20.328   
861               10.531            22.817               2.422   
862              607.000            30.902               6.000   
863               34.672            83.210              11.779   

     + 10 Mbps - 20 Mbps  + 20 Mbps - 30 Mbps  + 30 Mbps    OTROS      Total  
0                244.555               98.670  3.760.109   97.634  4.794.665  
1                 36.336                8.116  1.299.117  106.772  1.551.926  
2                  4.638                3.122     52.739    2.280     70.909  
3                  6.704                3.039     94.866    3.909    143.663  
4                 34.728               15.770     19.048   20.026    168.759  
..                   ...                  ...        ...      ...        ...  
859                1.000                0.000          0    0.000     26.760  
860                6.845               23.000        668    0.000    506.013  
861              109.000                0.000          0    0.000     37.113  
862                0.000                0.000          0    0.000     31.527  
863              362.000                3.000          0    0.000    130.032  

[864 rows x 12 columns]

In [28]:
acceso_internet_vel.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 864 entries, 0 to 863
Data columns (total 12 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Año                  864 non-null    int64  
 1   Trimestre            864 non-null    int64  
 2   Provincia            864 non-null    object 
 3   HASTA 512 kbps       864 non-null    float64
 4   + 512 Kbps - 1 Mbps  864 non-null    float64
 5   + 1 Mbps - 6 Mbps    864 non-null    object 
 6   + 6 Mbps - 10 Mbps   864 non-null    float64
 7   + 10 Mbps - 20 Mbps  864 non-null    float64
 8   + 20 Mbps - 30 Mbps  864 non-null    float64
 9   + 30 Mbps            864 non-null    object 
 10  OTROS                858 non-null    float64
 11  Total                864 non-null    object 
dtypes: float64(6), int64(2), object(4)
memory usage: 81.1+ KB


## **`internet_banda`**

Número de accesos al servicio de Internet fijo por banda ancha y banda angosta en cada provincia (trimestral)

In [29]:
internet_banda = pd.read_csv('../Telecommunications data analysis/Original datasets/Internet_BAF.csv')
internet_banda

Año  Trimestre            Provincia  Banda ancha fija  Dial up  Total
0    2022          4         Buenos Aires                 5      6.0      5
1    2022          4      Capital Federal                 2      2.0      2
2    2022          4            Catamarca                71      1.0     71
3    2022          4                Chaco               144      5.0    144
4    2022          4               Chubut               168    904.0    169
..    ...        ...                  ...               ...      ...    ...
859  2014          1           Santa Cruz                26    460.0     27
860  2014          1             Santa Fe               506    612.0    507
861  2014          1  Santiago Del Estero                37      9.0     37
862  2014          1     Tierra Del Fuego                27    766.0     28
863  2014          1              Tucumán               130     88.0    130

[864 rows x 6 columns]

In [30]:
internet_banda.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 864 entries, 0 to 863
Data columns (total 6 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Año               864 non-null    int64  
 1   Trimestre         864 non-null    int64  
 2   Provincia         864 non-null    object 
 3   Banda ancha fija  864 non-null    int64  
 4   Dial up           862 non-null    float64
 5   Total             864 non-null    int64  
dtypes: float64(1), int64(4), object(1)
memory usage: 40.6+ KB


## **`internet_por_loc`**
Listado de localidades con conectividad a internet, con detalle por tipo de conexión.

In [31]:
internet_por_loc = pd.read_csv('../Telecommunications data analysis/Original datasets/Listado_de_localidades_con_conectividad_a_internet.csv')
internet_por_loc

Provincia     Partido                                Localidad ADSL  \
0     BUENOS AIRES  9 de Julio  Alfredo Demarchi (Est. Facundo Quiroga)   SI   
1     BUENOS AIRES  9 de Julio                        Carlos María Naón   --   
2     BUENOS AIRES  9 de Julio                                 Dudignac   --   
3     BUENOS AIRES  9 de Julio                 La Aurora (Est. La Niña)   --   
4     BUENOS AIRES  9 de Julio           Manuel B. Gonnet (Est. French)   SI   
...            ...         ...                                      ...  ...   
4306  BUENOS AIRES       Pilar                                    Pilar   SI   
4307  BUENOS AIRES       Pilar                        Presidente Derqui   SI   
4308  BUENOS AIRES       Pilar                       Roberto De Vicenzo   --   
4309  BUENOS AIRES       Pilar                             Santa Teresa   --   
4310  BUENOS AIRES       Pilar                              Tortuguitas   --   

     Cablemódem Dial Up Fibra óptica  4G  3G Telefonía Fija Wireless Satelital  
0            --      --           --  SI  SI             SI       --        --  
1            --      --           --  SI  SI             SI       --        --  
2            --      SI           --  SI  SI             SI       SI        --  
3            --      --           --  SI  --             --       --        --  
4            --      --           SI  SI  SI             SI       --        --  
...         ...     ...          ...  ..  ..            ...      ...       ...  
4306         SI      SI           SI  SI  SI             SI       SI        SI  
4307         SI      --           SI  SI  SI             SI       SI        --  
4308         --      --           --  SI  SI             --       --        --  
4309         --      --           --  SI  SI             --       --        SI  
4310         --      --           SI  SI  SI             --       SI        SI  

[4311 rows x 12 columns]

In [32]:
internet_por_loc.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4311 entries, 0 to 4310
Data columns (total 12 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   Provincia       4311 non-null   object
 1   Partido         4311 non-null   object
 2   Localidad       4311 non-null   object
 3   ADSL            4311 non-null   object
 4   Cablemódem      4311 non-null   object
 5   Dial Up         4311 non-null   object
 6   Fibra óptica    4311 non-null   object
 7   4G              4311 non-null   object
 8   3G              4311 non-null   object
 9   Telefonía Fija  4311 non-null   object
 10  Wireless        4311 non-null   object
 11  Satelital       4311 non-null   object
dtypes: object(12)
memory usage: 404.3+ KB
